docs/api/locking.md

```diff
@@ -0,0 +1,449 @@
+response section to see what Lock properties are possible.
+* `next_cursor` - Optional string cursor that the server can return if there
+are more locks matching the given filters. The client will re-do the request,
+setting the `?cursor` query value with this `next_cursor` value.
+
+Note: If the server has no locks, it must return an empty `locks` array.
+
+```js
+// HTTP/1.1 200 Ok
+// Content-Type: application/vnd.git-lfs+json
+{
+  "locks": [
+    {
+      "id": "some-uuid",
+      "path": "/path/to/file",
+      "locked_at": "2016-05-17T15:49:06+00:00",
+      "committer": {
+        "name": "Jane Doe"
+      }
+    }
+  ],
+  "next_cursor": "optional next ID",
+}
+```
+
+### Unauthorized Response
+
+Lock servers should require that users have pull access to the repository before
+they can list locks.
+
+* `message` - String error message.
+* `request_id` - Optional String unique identifier for the request. Useful for
+debugging.
+* `documentation_url` - Optional String to give the user a place to report
+errors.
+
+```js
+// HTTP/1.1 403 Forbidden
+// Content-Type: application/vnd.git-lfs+json
+{
+  "message": "You must have pull access to list locks",
+  "documentation_url": "https://lfs-server.com/docs/errors",
+  "request_id": "123"
+}
+```
+
+### Error Response
+
+* `message` - String error message.
+* `request_id` - Optional String unique identifier for the request. Useful for
+debugging.
+* `documentation_url` - Optional String to give the user a place to report
+errors.
+
+```js
+// HTTP/1.1 500 Internal server error
+// Content-Type: application/vnd.git-lfs+json
+{
+  "message": "unable to list locks",
+  "documentation_url": "https://lfs-server.com/docs/errors",
+  "request_id": "123"
+}
+```
+
+## List Locks for Verification
+
+The client can use the Lock Verification endpoint to check for active locks
+that can affect a Git push. For a caller, this endpoint is very similar to the
+"List Locks" endpoint above, except:
+
+* Verification requires a `POST` request.
+* The `cursor` and `limit` values are sent as properties in the json request
+body.
+* The response includes locks partitioned into `ours` and `theirs` properties.
+
+Clients send the following to list locks for verification by sending a `POST`
+to `/locks/verify` (appended to the LFS server url, as described above):
+
+* `cursor`
+* `limit`
+
+```js
+// POST https://lfs-server.com/locks/verify
+// Accept: application/vnd.git-lfs+json
+// Content-Type: application/vnd.git-lfs+json
+// Authorization: Basic ...
+{
+  "cursor": "optional cursor",
+  "limit": 100 // also optional
+}
+```
+
+Note: As more advanced locking workflows are implemented, more details will
+likely be added to this request body in future iterations.
+
+### Successful Response
+
+A successful response will list the relevant locks:
+
+* `ours` - Array of Lock objects matching files that the user is allowed to
+modify.
+* `theirs` - Array of Lock objects matching files that the user is not allowed
```

Comment:
I think "allowed to modify" is ambiguous. Can we be more specific and say:

* ours - Array of lock objects currently owned by the authenticated user
* theirs - Array of lock objects currently owned by other users